implementations/rust/ockam/ockam_command/src/node/create.rs

```diff
@@ -81,9 +81,13 @@ pub struct CreateCommand {
     )]
     pub http_server: bool,
 
-    /// Enable the HTTP server at the given port.
-    #[arg(long, value_name = "PORT", conflicts_with = "http_server")]
-    pub http_server_port: Option<u16>,
+    /// Disable the node's status endpoint that serves the healthcheck endpoint.
+    #[arg(long, value_name = "BOOL", default_value_t = false)]
+    pub no_status_endpoint: bool,
+
+    /// Specify the port that the status endpoint will listen to.
+    #[arg(long, value_name = "PORT", conflicts_with = "no_status_endpoint")]
+    pub status_endpoint_port: Option<u16>,
```

Comment:
Could we parse to `Port` directly, with `Port::Explicit` if the port is given and `Port::TryExplicitOrRandom(DEFAULT_NODE_STATUS_ENDPOINT_PORT)` as a default value?

Reply:
It must be an option for when the `--no-status-endpoint` flag is passed. And if we set a default value, we'll still have to handle in the `parse_args` function the cases where either  `--no-status-endpoint` or `--status-endpoint-port` flags are passed.

Reply:
I can't find a way to use `default_value_t` with an optional argument. I'd leave it as it is for now. I've pushed a small change adding some `derives` to the `Port` struct.

Reply:
> It must be an option for when the --no-status-endpoint flag is passed

It doesn't have to, since we would discard the value later in `get_status_endpoint_port`. Yes, `Some` would work as well.

Reply:
Oh I see what you mean. Let me try that

Reply:
> It doesn't have to, since we would discard the value later in get_status_endpoint_port. Yes, Some would work as well.

I was trying that and I think it has to. If it's not an option, we must provide a default value. We also need a parser function.

```
/// Specify the port that the status endpoint will listen to.
    #[arg(long, value_name = "PORT", default_value_t = status_endpoint_default_value(), value_parser = parse_status_endpoint_port)]
    pub status_endpoint_port: Port,
```

In the parser function, we can't tell whether the given input is the default value (i.e. `node create` vs `node create --status-endpoint-port 23345`) which would lead to an unexpected behavior.

Reply:
I'm not sure I understand but I trust that you've carefully explored all options (no pun intended) now :-). Thanks!

Reply:
Just to clarify: in the example from above, we have an argument with a default value and a value parser:
- The default value is `Port::TryExplicitOrRandom(DEFAULT_NODE_STATUS_ENDPOINT_PORT)`: clap converts this to a string value using the Display implementation. That is, the actual default value is `"23345"`
- Then we have the parser, which will receive the `"23345"` string. At that point we no longer now whether that value was the default or the provided by the user, so we can't just return a `Port::Explicit`

Does that makes sense?